chore: disable automatic semantic release

arkaitzgarro opened the pull request:
## Proposed changes

I disabled automatic semantic release from CI, since it's not working properly, and it's a pain to make it work, tbh. And developers should always be able to publish a new version form their development machines.

As a bonus, I added the same contribution guidelines, code of conduct, templates etc that we already use on other WeTransfer OS projects.

## Types of changes

- [ ] Docs (missing or updated docs)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement / maintenance (removing technical debt, performance, tooling, etc...)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/WeTransfer/eslint-config-wetransfer/blob/master/.github/CONTRIBUTING.md) doc
- [x] I have added necessary documentation (if appropriate)
- [ ] Code changes are covered by unit tests.
- [ ] Any dependent changes have been merged and published in downstream modules

seelts commented:
@arkaitzgarro 
I think it was definitely possible to split this in two PRs. Those templates and guidelines are not related to the PR's subject actually.

Anyway, regarding the subject. Why do you want to disable CI?
I'm, fine with adding the ability to release manually. But it would be also awesome to be able to have automated CI release flow.
Also where is `npm publish` now?

arkaitzgarro commented:
@seelts I removed the unrelated templates, and will create a new PR.

I removed automatic release because of several reasons:
- It's a pain to setup for CI. It was working for Travis, but since we moved to CircleCI, we need to make it work again. It will be the case in the future if migrate to another CI.
- It is a potential security issue to have npm and GitHub credentials with publishing and commit permissions on a third party service.
- The releases will always be under the same name, which is not nice for the rest of developers.

About `npm publish`, `npm run release` will do all the magic for you. Analyse commits, publish to npm, create a tag, create a release on GH, etc.
